### PR TITLE
🐛 inabox: support sending positions to ads with null origins

### PR DESCRIPTION
**Pull request by jeffkaufman**
Google AdManager is interested in experimenting with loading AMP Inabox ads with a null origin, for better performance and security.  Before this commit, an inabox ad with a null origin couldn't subscribe to position tracking, because you can't send a postmessage restricted to a null origin.  Reading the code, though, there's no reason to be restricting what origin we send the positions to, because (a) we've already verified that this iframe is whitelisted and allowed to learn its position, and (b) we're post messaging back directly to the requesting frame.

Switch to a targetOrigin of `*` to indicate that we have no preference on the receiving origin.

Also add a simple implementation of data-url rendering to the visual tests, so we can verify it works.  Before the change to `InaboxMessagingHost` the test gives a `Invalid target origin 'null' in a call to 'postMessage'.` console error.

### CODE REVIEW

File: ads/inabox/inabox-messaging-host.js [hunk 1]
```diff
@@ -200,11 +198,10 @@ export class InaboxMessagingHost {
    * @param {string} origin
```
Comment (zombifier) [inline ads/inabox/inabox-messaging-host.js]: Remove this line.

Comment (zombifier) [reply ads/inabox/inabox-messaging-host.js]: Also linter is complaining about some reformatting: https://travis-ci.org/ampproject/amphtml/jobs/608264619?utm_medium=notification&utm_source=github_status

File: ads/inabox/inabox-messaging-host.js [hunk 2]
```diff
@@ -197,14 +191,13 @@ export class InaboxMessagingHost {
    *
    * @param {!Object} request
    * @param {!Window} source
-   * @param {string} origin
    * @param {?JsonObject} data
    */
-  sendPosition_(request, source, origin, data) {
+  sendPosition_(request, source, data) {
```
Comment (lannka) [inline ads/inabox/inabox-messaging-host.js]: shall we still pass down `origin` and do `*` only when it is `null`?

cc @molnarg to chime in from security's perspective

File: ads/inabox/inabox-messaging-host.js [hunk 3]
```diff
@@ -197,14 +191,13 @@ export class InaboxMessagingHost {
...
     dev().fine(TAG, 'Sent position data to [%s] %s', request.sentinel, data);
     source./*OK*/ postMessage(
       serializeMessage(MessageType.POSITION, request.sentinel, data),
-      origin
+      '*'
```
Comment (zhouyx) [inline ads/inabox/inabox-messaging-host.js]: This works well as long as we have the request sentinel. The only concern as @lannka mentioned is that the sentinel will be shared to other origins. LGTM as long as we are good with that.

File: ads/inabox/inabox-messaging-host.js (hunk 2 above)
Comment (molnarg) [reply ads/inabox/inabox-messaging-host.js]: Based on the PR description, it's public data so it looks fine to leave it like this. Please add a comment that explains it.

Comment (jeffkaufman) [reply ads/inabox/inabox-messaging-host.js]: added comment

**Comment by jeffkaufman** [comment]
It turns out that the Percy visual diff tests can't handle cross-domain iFrames: https://percy.io/ampproject/amphtml/builds/3117157 After debugging this with @zombifier it doesn't sound like this is something current infrastructure can support, and the whole goal of the test is verifying cross-domain behavior.

I've written a Google Ads internal test that will alert us if this stops working on RC, but that's later than I'd like to find out about problems running cross-domain.

**Comment by jeffkaufman** [comment]
The Travis failure is an unrelated one in extensions/amp-list/0.1/amp-list.js

File: ads/inabox/inabox-messaging-host.js (hunk 2 above)
Comment (lannka) [reply ads/inabox/inabox-messaging-host.js]: @molnarg still want to throw my 2c. 

I don't think user's scroll position data are really public data. We do have some mechanism in place to make sure only iframes created by the ads tag on the page can request the data. we also use sentinel token to make sure the integrity in those iframes.

My own principal for security-wise thing is the tighter constraint the better, if it does not come with big trade-off.

Comment (jeffkaufman) [reply ads/inabox/inabox-messaging-host.js]: @lannka I can't tell from your comment whether you're ok with this PR?

Comment (lannka) [reply ads/inabox/inabox-messaging-host.js]: I'm not convinced that we should remove this extra check as the assumption that "data is public" is wrong.

**Comment by jeffkaufman** [comment]
@lannka it looks to me like the percy diffs are unrelated; can this still be merged?

File: ads/inabox/inabox-messaging-host.js (hunk 2 above)
Comment (lannka) [reply ads/inabox/inabox-messaging-host.js]: Chat with @jeffkaufman offline and I am convinced there is no extra security benefit here to set the `origin`.

**Pull request merged**